Documentation/README.md

```diff
@@ -48,23 +48,140 @@ class MyCustomService: RestfulRead {
+    var firstName: String
+    var lastName: String
+}
+
+struct Response: Decodable {
+    var location: String
+    var firstName: String
+    var lastName: String
+}
+
+class MyCustomService: RestfulWrite {
+    
+    private let domain: URL = URL(string: "https://mydomain.com/entities")!
+    
+    public func createWithUrlEncodingAndExtractResponse(entity: Request) -> Promise<Response> {
+        return entity.toPromisedQueryParameters(in: domain).then(write)
+    }
+    
+    public func createWithUrlEncodingAndExtractLocation(entity: Request) -> Promise<String> {
+        return entity.toPromisedQueryParameters(in: domain).then(writeAndExtractLocation)
+    }
+    
+    public func createWithJsonEncodingAndExtractResponse(entity: Request) -> Promise<Response> {
+        return write(domain, entity)
+    }
+    
+    public func createWithJsonEncodingAndExtractLocation(entity: Request) -> Promise<String> {
+        return writeAndExtractLocation(domain, entity)
+    }
+    
+    public func createWithJsonEncodingAndExtractHeaders(entity: Request) -> Promise<HTTPHeadersType> {
+        let headerKeys: [String] = ["Location", "Transaction"]
+        return writeAndExtractHeaders(domain, entity, headerKeys)
+    }
+}
+```
+
 ### RestfulWriteError Protocol - for executing POST requests with server validation response <a name="restfulwriteerror"></a>
-TODO
+This protocol provides possibility to execute POST requests on an URL and returns a promise with a decodable result or a location header.
+In addition to the RestfulWrite Protocol a decodable error can be validated.
+
+[For services and examples please look at the RestfulWrite Protocol](#restfulwrite)
+
 ### RestfulUpdate Protocol - for executing PUT requests <a name="restfulupdate"></a>
-TODO
+This protocol provides possibility to execute PUT requests on an URL and returns an empty promise or a promise with the updated entity.
+
+Services:
+
+```swift
+func update(_ url: URL, _ entity: Encodable) -> Promise<Void>
+
+func update(_ url: URL, _ entity: Encodable) -> Promise<Decodable>
+```
+
+Examples:
+
+```swift
+struct Entity: Codable {
+    var firstName: String
+    var lastName: String
+}
+
+class MyCustomService: RestfulUpdate {
+
+    private let domain: URL = URL(string: "https://mydomain.com/entities")!
+    
+    public func change(entity: Entity) -> Promise<Void> {
+        return update(domain, entity)
+    }
+    
+    public func change(entity: Entity) -> Promise<Entity> {
+        return update(domain, entity)
+    }
+}
+```
+
 ### RestfulUpdateError Protocol - for executing PUT requests with server validation response <a name="restfulupdateerror"></a>
-TODO
-### RestfulDelete Protocol - for executing PUT requests <a name="restfuldelete"></a>
-TODO
+This protocol provides possibility to execute PUT requests on an URL and returns an empty promise or a promise with the updated entity.
+In addition to the RestfulUpdate Protocol a decodable error can be validated.
```

Comment:
the description is not quite unterstandable. Suggestion: "Additionally to the RestfulUpdate protocol, this protocol can decode server errors by providing the associated decodable error type."